providers/oauth/oauth.webrequest.js

```diff
@@ -42,26 +42,29 @@ ChromeWebRequestAuth.prototype.initiateOAuth = function(redirectURIs, continuati
   return false;
 };
 
-ChromeWebRequestAuth.prototype.launchAuthFlow = function(authUrl, stateObj, continuation) {
+ChromeWebRequestAuth.prototype.launchAuthFlow = function(authUrl, stateObj, interactive, continuation) {
   "use strict";
+  if (interactive === undefined) {
+    interactive = true;
+  }
   var listener = this.reqListener.bind(this, stateObj, continuation);
   this.listeners[stateObj.state] = listener;
   chrome.webRequest.onBeforeRequest.addListener(listener, {
     types: ["main_frame"],
     urls: [stateObj.redirect]
   });
-  
+
   chrome.tabs.create({
     url: authUrl,
-    active: true
+    active: interactive
```

Comment:
Cool! Thanks for the background info. this definitely helps my mental model here.

Related question then:
If this behavior depends on you already being logged into Facebook and already granted uproxy access. In the case where that's not the case, you're just going to have a dead tab in the background. Can we check for that, cleanup if it's not going anywhere, and properly reject the call the launchAuthFlow?